network_wrangler/scenario.py

```diff
@@ -38,627 +41,530 @@ class Scenario(object):
-        has_conflict_error: boolean indicating if there is are conflicting project cards
-        prerequisites_sorted: boolean indicating if the project cards have
-            been sorted to make sure cards that are pre-requisites are applied first
     """
 
-    def __init__(self, base_scenario: dict, project_cards: List[ProjectCard] = None):
+    def __init__(
+        self,
+        base_scenario: Union[Scenario, dict],
+        project_card_list: list[ProjectCard] = None,
+        name="",
+    ):
         """
         Constructor
 
         args:
         base_scenario: dict the base scenario
-        project_cards: list this scenario's project cards
+        project_card_list: list of ProjectCard instances
         """
+        WranglerLogger.info(
+            f"Creating Scenario with {len(project_card_list)} project cards"
+        )
 
-        self.road_net = None
-        self.transit_net = None
+        if type(base_scenario) == "Scenario":
+            base_scenario = base_scenario.__dict__
 
         self.base_scenario = base_scenario
-
+        self.name = name
         # if the base scenario had roadway or transit networks, use them as the basis.
-        if self.base_scenario.get("road_net"):
-            self.road_net = copy.deepcopy(self.base_scenario["road_net"])
-        if self.base_scenario.get("transit_net"):
-            self.transit_net = copy.deepcopy(self.base_scenario["transit_net"])
-
-        # if the base scenario had applied projects, add them to the list of applied
-        self.applied_projects = []
-        if self.base_scenario.get("applied_projects"):
-            self.applied_projects = base_scenario["applied_projects"]
+        self.road_net = copy.deepcopy(self.base_scenario.get("road_net"))
+        self.transit_net = copy.deepcopy(self.base_scenario.get("transit_net"))
 
-        self.project_cards = project_cards
-        self.ordered_project_cards = OrderedDict()
+        self.project_cards = {}
+        self._planned_projects = []
+        self._queued_projects = None
+        self.applied_projects = self.base_scenario.get("applied_projects", [])
 
-        self.prerequisites = {}
-        self.corequisites = {}
-        self.conflicts = {}
+        self.prerequisites = self.base_scenario.get("prerequisites", {})
+        self.corequisites = self.base_scenario.get("corequisites", {})
+        self.conflicts = self.base_scenario.get("conflicts", {})
 
-        self.requisites_checked = False
-        self.conflicts_checked = False
+        for p in project_card_list:
+            self._add_project(p)
 
-        self.has_requisite_error = False
-        self.has_conflict_error = False
+    @property
+    def projects(self):
+        return self.applied_projects + self.queued_projects
 
-        self.prerequisites_sorted = False
+    @property
+    def queued_projects(self):
+        """Returns a list version of _queued_projects queue."""
+        if self._queued_projects is None:
```

Comment:
This IF condition is never True in my tests, `self._queue_projects = deque([])` somehow.

Reply:
ah yes, i can just check the truthiness/bool version of it instead... fixed!